src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -986,7 +986,7 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                         body = (BoundBlock)body.WithHasErrors();
                     }
 
-                    if (body != null && methodSymbol.IsConstructor())
+                    if (body != null && methodSymbol.IsConstructor() && !(methodSymbol is SynthesizedRecordConstructor))
```

Comment:
> && !(methodSymbol is SynthesizedRecordConstructor) [](start = 68, length = 51)

Could you please provide some details about this change? I would not expect us to suppress any analysis for primary constructor.

Reply:
It's not essential (optimization). I'll remove

---
In reply to: [454449395](https://github.com/dotnet/roslyn/pull/45898#discussion_r454449395) [](ancestors = 454449395)